frontend-react/src/hooks/network/Organizations/PublicKeys/UseCreateOrganizationPublicKey.ts

```diff
@@ -0,0 +1,52 @@
+import { useCallback } from "react";
+import { UseMutationResult } from "@tanstack/react-query";
+
+import {
+    RSApiKeysResponse,
+    servicesEndpoints,
+} from "../../../../config/endpoints/settings";
+import { useAuthorizedMutationFetch } from "../../../../contexts/AuthorizedFetchContext";
+import { useSessionContext } from "../../../../contexts/SessionContext";
+
+export interface OrganizationPublicKeyPostArgs {
+    kid: string;
+    sender: string;
+}
+
+export type UseCreateOrganizationPublicKeyResult = UseMutationResult<
+    RSApiKeysResponse,
+    unknown,
+    OrganizationPublicKeyPostArgs
+>;
+
+export default function useCreateOrganizationPublicKey(): UseCreateOrganizationPublicKeyResult {
+    const { activeMembership } = useSessionContext();
+    const parsedName = activeMembership?.parsedName;
+
+    const { authorizedFetch, rsUseMutation } = useAuthorizedMutationFetch<
+        RSApiKeysResponse,
+        unknown,
+        OrganizationPublicKeyPostArgs
+    >();
+    const mutationFn = useCallback(
```

Comment:
I really don't think we need to be using `useCallback` and `useMemo` so frequently; if the references for `authorizedFetch` or `parsedName` change, we should trigger a re-render anyway, no?

Not a huge deal, but I just don't think we need to be so focused on memoization when re-renders are expected behaviors anyhow.

Reply:
I'm likely guilty of this as well 😆. It's true that the re-creation of simple functions is low performance lift.

Reply:
@jpandersen87 any thoughts on Stephens comment? This was following the new pattern from https://github.com/CDCgov/prime-reportstream/pull/8690

Reply:
Completely OK to drop the callback. I need to do the same.